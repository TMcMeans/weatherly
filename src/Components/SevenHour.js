import React from 'react';
import Card from "./Card";

function SevenHour(props)  {
  const sevenHourData = props.hourlyForecast.slice(0, 7);
  return ( 
     <div className="sevenHourCards">
       {sevenHourData.map( (data, index) => {
          return(
              <Card {...data} key={index} />
            )
       })}
     </div> 
   );
}

export default SevenHour;
