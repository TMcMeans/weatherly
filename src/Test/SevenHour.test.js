import React from "react";
import renderer from 'react-test-renderer';
import { shallow, mount } from "enzyme";
import SevenHour from "../Components/SevenHour";
import data from "../mockAPI.js";

describe("SevenHour component", () => {
  let wrapper;

  // beforeEach(() => {
  //   wrapper = shallow(<SevenHour />);
  //   localStorage.clear();
  // });

  it.skip("should be a thing", () => {
    expect(wrapper).toBeDefined();
  });

  let { hourly_forecast } = data

  it('renders correctly', () => {
  const tree = renderer.create(
    <SevenHour hourlyForecast={ hourly_forecast }/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
  });

});

