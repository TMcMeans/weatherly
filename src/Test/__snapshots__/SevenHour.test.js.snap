// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SevenHour component renders correctly 1`] = `
<div
  className="sevenHourCards"
>
  <div
    className="card"
  >
    <h1>
      12 am
    </h1>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-cloud fa-w-20 "
      data-icon="cloud"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M537.6 226.6c4.1-10.7 6.4-22.4 6.4-34.6 0-53-43-96-96-96-19.7 0-38.1 6-53.3 16.2C367 64.2 315.3 32 256 32c-88.4 0-160 71.6-160 160 0 2.7.1 5.4.2 8.1C40.2 219.8 0 273.2 0 336c0 79.5 64.5 144 144 144h368c70.7 0 128-57.3 128-128 0-61.9-44-113.6-102.4-125.4z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <h1>
      47
      ° F
    </h1>
    <p>
      Partly Cloudy
    </p>
    <p>
      
    </p>
  </div>
  <div
    className="card"
  >
    <h1>
      1 pm
    </h1>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-cloud fa-w-20 "
      data-icon="cloud"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M537.6 226.6c4.1-10.7 6.4-22.4 6.4-34.6 0-53-43-96-96-96-19.7 0-38.1 6-53.3 16.2C367 64.2 315.3 32 256 32c-88.4 0-160 71.6-160 160 0 2.7.1 5.4.2 8.1C40.2 219.8 0 273.2 0 336c0 79.5 64.5 144 144 144h368c70.7 0 128-57.3 128-128 0-61.9-44-113.6-102.4-125.4z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <h1>
      49
      ° F
    </h1>
    <p>
      Partly Cloudy
    </p>
    <p>
      
    </p>
  </div>
  <div
    className="card"
  >
    <h1>
      2 pm
    </h1>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-cloud fa-w-20 "
      data-icon="cloud"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M537.6 226.6c4.1-10.7 6.4-22.4 6.4-34.6 0-53-43-96-96-96-19.7 0-38.1 6-53.3 16.2C367 64.2 315.3 32 256 32c-88.4 0-160 71.6-160 160 0 2.7.1 5.4.2 8.1C40.2 219.8 0 273.2 0 336c0 79.5 64.5 144 144 144h368c70.7 0 128-57.3 128-128 0-61.9-44-113.6-102.4-125.4z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <h1>
      49
      ° F
    </h1>
    <p>
      Partly Cloudy
    </p>
    <p>
      
    </p>
  </div>
  <div
    className="card"
  >
    <h1>
      3 pm
    </h1>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-sun fa-w-16 "
      data-icon="sun"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M274.835 12.646l25.516 62.393c4.213 10.301 16.671 14.349 26.134 8.492l57.316-35.479c15.49-9.588 34.808 4.447 30.475 22.142l-16.03 65.475c-2.647 10.81 5.053 21.408 16.152 22.231l67.224 4.987c18.167 1.348 25.546 24.057 11.641 35.826L441.81 242.26c-8.495 7.19-8.495 20.289 0 27.479l51.454 43.548c13.906 11.769 6.527 34.478-11.641 35.826l-67.224 4.987c-11.099.823-18.799 11.421-16.152 22.231l16.03 65.475c4.332 17.695-14.986 31.73-30.475 22.142l-57.316-35.479c-9.463-5.858-21.922-1.81-26.134 8.492l-25.516 62.393c-6.896 16.862-30.774 16.862-37.67 0l-25.516-62.393c-4.213-10.301-16.671-14.349-26.134-8.492l-57.317 35.479c-15.49 9.588-34.808-4.447-30.475-22.142l16.03-65.475c2.647-10.81-5.053-21.408-16.152-22.231l-67.224-4.987c-18.167-1.348-25.546-24.057-11.641-35.826L70.19 269.74c8.495-7.19 8.495-20.289 0-27.479l-51.454-43.548c-13.906-11.769-6.527-34.478 11.641-35.826l67.224-4.987c11.099-.823 18.799-11.421 16.152-22.231l-16.03-65.475c-4.332-17.695 14.986-31.73 30.475-22.142l57.317 35.479c9.463 5.858 21.921 1.81 26.134-8.492l25.516-62.393c6.896-16.861 30.774-16.861 37.67 0zM392 256c0-74.991-61.01-136-136-136-74.991 0-136 61.009-136 136s61.009 136 136 136c74.99 0 136-61.009 136-136zm-32 0c0 57.346-46.654 104-104 104s-104-46.654-104-104 46.654-104 104-104 104 46.654 104 104z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <h1>
      51
      ° F
    </h1>
    <p>
      Clear
    </p>
    <p>
      
    </p>
  </div>
  <div
    className="card"
  >
    <h1>
      4 pm
    </h1>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-sun fa-w-16 "
      data-icon="sun"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M274.835 12.646l25.516 62.393c4.213 10.301 16.671 14.349 26.134 8.492l57.316-35.479c15.49-9.588 34.808 4.447 30.475 22.142l-16.03 65.475c-2.647 10.81 5.053 21.408 16.152 22.231l67.224 4.987c18.167 1.348 25.546 24.057 11.641 35.826L441.81 242.26c-8.495 7.19-8.495 20.289 0 27.479l51.454 43.548c13.906 11.769 6.527 34.478-11.641 35.826l-67.224 4.987c-11.099.823-18.799 11.421-16.152 22.231l16.03 65.475c4.332 17.695-14.986 31.73-30.475 22.142l-57.316-35.479c-9.463-5.858-21.922-1.81-26.134 8.492l-25.516 62.393c-6.896 16.862-30.774 16.862-37.67 0l-25.516-62.393c-4.213-10.301-16.671-14.349-26.134-8.492l-57.317 35.479c-15.49 9.588-34.808-4.447-30.475-22.142l16.03-65.475c2.647-10.81-5.053-21.408-16.152-22.231l-67.224-4.987c-18.167-1.348-25.546-24.057-11.641-35.826L70.19 269.74c8.495-7.19 8.495-20.289 0-27.479l-51.454-43.548c-13.906-11.769-6.527-34.478 11.641-35.826l67.224-4.987c11.099-.823 18.799-11.421 16.152-22.231l-16.03-65.475c-4.332-17.695 14.986-31.73 30.475-22.142l57.317 35.479c9.463 5.858 21.921 1.81 26.134-8.492l25.516-62.393c6.896-16.861 30.774-16.861 37.67 0zM392 256c0-74.991-61.01-136-136-136-74.991 0-136 61.009-136 136s61.009 136 136 136c74.99 0 136-61.009 136-136zm-32 0c0 57.346-46.654 104-104 104s-104-46.654-104-104 46.654-104 104-104 104 46.654 104 104z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <h1>
      50
      ° F
    </h1>
    <p>
      Clear
    </p>
    <p>
      
    </p>
  </div>
  <div
    className="card"
  >
    <h1>
      5 pm
    </h1>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-sun fa-w-16 "
      data-icon="sun"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M274.835 12.646l25.516 62.393c4.213 10.301 16.671 14.349 26.134 8.492l57.316-35.479c15.49-9.588 34.808 4.447 30.475 22.142l-16.03 65.475c-2.647 10.81 5.053 21.408 16.152 22.231l67.224 4.987c18.167 1.348 25.546 24.057 11.641 35.826L441.81 242.26c-8.495 7.19-8.495 20.289 0 27.479l51.454 43.548c13.906 11.769 6.527 34.478-11.641 35.826l-67.224 4.987c-11.099.823-18.799 11.421-16.152 22.231l16.03 65.475c4.332 17.695-14.986 31.73-30.475 22.142l-57.316-35.479c-9.463-5.858-21.922-1.81-26.134 8.492l-25.516 62.393c-6.896 16.862-30.774 16.862-37.67 0l-25.516-62.393c-4.213-10.301-16.671-14.349-26.134-8.492l-57.317 35.479c-15.49 9.588-34.808-4.447-30.475-22.142l16.03-65.475c2.647-10.81-5.053-21.408-16.152-22.231l-67.224-4.987c-18.167-1.348-25.546-24.057-11.641-35.826L70.19 269.74c8.495-7.19 8.495-20.289 0-27.479l-51.454-43.548c-13.906-11.769-6.527-34.478 11.641-35.826l67.224-4.987c11.099-.823 18.799-11.421 16.152-22.231l-16.03-65.475c-4.332-17.695 14.986-31.73 30.475-22.142l57.317 35.479c9.463 5.858 21.921 1.81 26.134-8.492l25.516-62.393c6.896-16.861 30.774-16.861 37.67 0zM392 256c0-74.991-61.01-136-136-136-74.991 0-136 61.009-136 136s61.009 136 136 136c74.99 0 136-61.009 136-136zm-32 0c0 57.346-46.654 104-104 104s-104-46.654-104-104 46.654-104 104-104 104 46.654 104 104z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <h1>
      48
      ° F
    </h1>
    <p>
      Clear
    </p>
    <p>
      
    </p>
  </div>
  <div
    className="card"
  >
    <h1>
      6 pm
    </h1>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-sun fa-w-16 "
      data-icon="sun"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M274.835 12.646l25.516 62.393c4.213 10.301 16.671 14.349 26.134 8.492l57.316-35.479c15.49-9.588 34.808 4.447 30.475 22.142l-16.03 65.475c-2.647 10.81 5.053 21.408 16.152 22.231l67.224 4.987c18.167 1.348 25.546 24.057 11.641 35.826L441.81 242.26c-8.495 7.19-8.495 20.289 0 27.479l51.454 43.548c13.906 11.769 6.527 34.478-11.641 35.826l-67.224 4.987c-11.099.823-18.799 11.421-16.152 22.231l16.03 65.475c4.332 17.695-14.986 31.73-30.475 22.142l-57.316-35.479c-9.463-5.858-21.922-1.81-26.134 8.492l-25.516 62.393c-6.896 16.862-30.774 16.862-37.67 0l-25.516-62.393c-4.213-10.301-16.671-14.349-26.134-8.492l-57.317 35.479c-15.49 9.588-34.808-4.447-30.475-22.142l16.03-65.475c2.647-10.81-5.053-21.408-16.152-22.231l-67.224-4.987c-18.167-1.348-25.546-24.057-11.641-35.826L70.19 269.74c8.495-7.19 8.495-20.289 0-27.479l-51.454-43.548c-13.906-11.769-6.527-34.478 11.641-35.826l67.224-4.987c11.099-.823 18.799-11.421 16.152-22.231l-16.03-65.475c-4.332-17.695 14.986-31.73 30.475-22.142l57.317 35.479c9.463 5.858 21.921 1.81 26.134-8.492l25.516-62.393c6.896-16.861 30.774-16.861 37.67 0zM392 256c0-74.991-61.01-136-136-136-74.991 0-136 61.009-136 136s61.009 136 136 136c74.99 0 136-61.009 136-136zm-32 0c0 57.346-46.654 104-104 104s-104-46.654-104-104 46.654-104 104-104 104 46.654 104 104z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <h1>
      45
      ° F
    </h1>
    <p>
      Clear
    </p>
    <p>
      
    </p>
  </div>
</div>
`;
